adapters/huaweiads/huaweiads.go

```diff
@@ -278,14 +296,41 @@ func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest,
 	header = getHeaders(huaweiAdsImpExt, openRTBRequest, isTestAuthorization)
 	bidRequest := &adapters.RequestData{
 		Method:  http.MethodPost,
-		Uri:     a.endpoint,
+		Uri:     getFinalEndPoint(countryCode, a.endpoint),
 		Body:    reqJSON,
 		Headers: header,
 	}
 
 	return []*adapters.RequestData{bidRequest}, nil
 }
 
+// countryCode is alpha2, choose the corresponding site end point
+func getFinalEndPoint(countryCode string, defaultEndpoint string) string {
+	if countryCode == "" || len(countryCode) > 2 {
+		return defaultEndpoint
+	}
+
+	var europeanSiteCountryCodeGroup = map[string]string{"AX": "1", "AL": "1", "AD": "1", "AU": "1", "AT": "1", "BE": "1",
```

Comment:
It is slightly more efficient to define these as `map[string]interface{}`, as empty interfaces can be the most efficient to store. Also makes it a little more obvious that the value stored in the map has no actual use, it just exists to take advantage of the efficiency of the map's key lookup mechanics.

Reply:
Is this attempting to simulate a hash set? If so, it's most efficient to use `map[string]struct{}` as the empty struct consumes literally no memory. All the fun idiosyncrasies of Go. See: https://medium.com/@l.peppoloni/how-to-improve-your-go-code-with-empty-structs-3bd0c66bc531

Reply:
Thanks Scott, I was thinking empty structs but somehow landed on interfaces.

Reply:
Thank you all, it has been revised